src/Agent.Listener/CommandSettings.cs

```diff
@@ -38,6 +72,31 @@ public CommandSettings(IHostContext context, string[] args)
             _parser.Parse(args);
         }
 
+        // Validate commandline parser result
+        public void Validation()
+        {
+            // detect unknown commands
```

Comment:
I think the nested foreach is good considering it's a small number in the list (e.g. 10s) if it were 1000s i would be concerned.